### ISSUE TITLE
Driver fails to activate Monterey 12.2 Beta 21D5025f, while SIP is disabled.

### ISSUE THREAD

**Issue by Ajaymamtora** [pqrs-org/Karabiner-Elements#2947]
I've tried all the various fixes involving rebooting, allowing the app in security settings etc, but while SIP is disabled, it just will not work. I get this in the logs, endlessly:

`Karabiner-DriverKit-VirtualHIDDeviceClient expected: 10600, actual dext: std::nullopt`

I am trying this on an M1 Pro MBP.

**Comment by nkaretnikov** [comment]
yep, i confirm this. works fine on 12.0.1 with sip _enabled_. once i _disable_ sip, it breaks (driver is not loaded/cannot request permissions). so i just uninstalled the program and remapped via hidutil

https://developer.apple.com/library/archive/technotes/tn2450/_index.html

**Comment by Muirium** [comment]
Apple advises only disabling SIP briefly, when you really need to:

https://developer.apple.com/documentation/security/disabling_and_enabling_system_integrity_protection

I've installed [Audio Hijack](https://rogueamoeba.com/audiohijack/), which requires briefly disabling SIP to install its audio engine, but my M1 Air runs with SIP enabled (just checked by running `csrutil status`) and Karabiner works just fine.

I don't know what the use case is for leaving SIP disabled, full time.

**Comment by nkaretnikov** [comment]
i like to deliberately disable it since some low-level programs that i'm using require this. i'm aware of the risks tho

**Comment by Ajaymamtora** [comment]
I disabled SIP so I could use cDock.

**Comment by Ajaymamtora** [comment]
@Muirium I disabled SIP so I could use cDock, which requires it to be disabled full time, while using it.
